_posts/2018-05-07-A-Machine-Learning-Example-For-Business.md

```diff
@@ -0,0 +1,271 @@
+
+Because I had the power to create the data for this example, all 6 predictions were correct. For the second record the probability for both being late and not being late was 50%. If this would have been interpreted as `Yes`, the prediction would have been incorrect, and the accuracy would have fallen to 83.3% or 5/6.
+
+![Naive bayes prediction 50-50](/img/a-machine-learning-example-for-business/naive-bayes-prediction-50-50.png)
+
+## Evaluating the business benefit
+So far we have:
+1. Identified a business problem.
+2. Combined and cleaned possibly relevant data.
+3. Selected a suitable algorithm.
+4. Trained a model.
+5. Tested the model with historical data.
+6. Evaluated the prediction accuracy.
+
+Now it's time to evaluate the business benefit.
+
+We are confident, that our model is between 83.3% to 100% accurate for unseen observations, where only order day, order size and sub contractor are known before setting up the order. Let's trust the mid point and say that accuracy is 91.7%.
+
+You can count that in test data 3/6 or 50% of orders arrived late. Supposing that you would have used your trained machine learning model, you could have made the order only for those 3 or 4 deliveries that are predicted to arrive on time and a couple of additional orders that are not described in the test data.
+
+In real world the situation would be more complex, but you get the idea that accuracy of 91.7% is more than 50%. With these numbers you are probably also able to estimate the impact in terms of money.
+
+## Implementing the model to daily work
+
+Here are some ways to harness the solution for business.
+
+<b>Data study</b>. Improve company's process by experimenting the optimal order types.
+
+<b>Calculator</b>. Let employers to enter the three predictor variables to a web calculator before placing the order to evaluate the risk of receiving products late.
+
+<b>Automated orders</b>. Embed the calculation as part of the company's IT infrastructure. The system can make the optimal orders automatically.
+
+As time goes by, more historical orders can be used in training and testing. More data makes the model more reliable.
+
+The model can be trained with the order data from past year once a day for example. Frequent training and limiting data only to latest observations doesn't make machine literally <i>learn</i>, but rather this adjust the model to give the best guess for the given situation.
+
+## Summary
+A predictive machine learning model offers a way to see the future with the expectation that the environment remains somewhat unchanged. The business decisions don't need to be based on hunches and prediction methods can be evaluated systematically.
+
+Thoughtfully chosen machine learning algorithm is able to make optimal choices in routine tasks with much greater accuracy than humans. Machine learning makes it also possible to estimate the monetary impacts of decisions even before the actions has been taken.
```

Comment:
Here, I would probably speak of a thoughtfully chosen machine learning _model_ which is more important (e.g. what features you use et cetera) than the actual choice of the algorithm, which often is not actually that important.

Reply:
Ok.